classes/class.ilObjBigBlueButtonGUI.php

```diff
@@ -359,12 +359,17 @@ public function showContent()
             $my_tpl->setVariable("hasMeetingRecordings", $this->has_meeting_recordings  && boolval($values["choose_recording"]) ? "true" : "false");
 
             $bbbURL=$BBBHelper->joinURLModerator($this->object);
-        } else {
+        } 
+        else {
             $my_tpl = new ilTemplate("./Customizing/global/plugins/Services/Repository/RepositoryObject/BigBlueButton/templates/tpl.BigBlueButtonClient.html", true, true);
 
             $my_tpl->setVariable("classNotStartedText", $this->txt("class_not_started_yet"));
-
-            $bbbURL=$BBBHelper->joinURL($this->object);
+           
+                $my_tpl->setVariable("recordings", $this->buildRecordingUI());
```

Comment:
You should refactor the whole part. This is only copied fromt he "moderator" section. But in the moderator section, it will be checked, if int he currect BBB object recordings are allowed. This check is missing in your changed.

See:
https://github.com/Minervis-GmbH/BigBlueButton-Ilias-Plugin/blob/8fba2f4f5e05655429f37f406aeb6f8b8ff15afc/classes/class.ilObjBigBlueButtonGUI.php#L351-L358

Reply:
Yes that's right. But it also controls if the recordings ui is visible or no. With your change, users can see the recording ui when recordings in the bbb object are disallowed. So i think is should be checked if recordings are allowed.